### ISSUE TITLE
Certain epubs only read the first page 

### ISSUE THREAD

**Issue by luky92** [DrewThomasson/ebook2audiobook#146]
for some epubs only the first page is read you can download a sample epub that does this here:
https://wolnelektury.pl/katalog/lektura/brzydkie-kaczatko/
language is polish 
let me know if you need any help or debug logs im happy to provide these 
( note this ebook is in public domain just in case thats important)

**Comment by ROBERT-MCDOWELL** [comment]
the link you provided is a sample of the book right?

**Comment by luky92** [comment]
no it is a full "book" I assure its not only one page:
screenshhot of an epub reader view:
![image](https://github.com/user-attachments/assets/0f719a01-5729-4228-a8ea-c53ee9d4d11d)

**Comment by ROBERT-MCDOWELL** [comment]
when I edit the book with calibre, I can see only part1.xhtml as a part of the epub with some real text right?
other pages are fund1, fund2 etc... and fund1 contains
"Książka, którą czytasz, pochodzi z biblioteki fundacji [Wolne Lektury](https://wolnelektury.pl/). Naszą misją jest wspieranie dzieciaków w dostępie do lektur szkolnych oraz zachęcanie ich do czytania. Miło Cię poznać!"
is it this page it is conveted only?

**Comment by luky92** [comment]
yes you are right part 1 is the whole book the resto of it is like title page and some founding stuff

**Comment by ROBERT-MCDOWELL** [comment]
here is the issue: in the ebook world there is absolutely no rules of how to name or separate the real book text from legal, annotations, technical text, becoming very difficult to filter the ebook pages to just keep the real text of the book and avoid the A.I speech to say all the text from technical stuff which is very ennoying and burden to hear.
for now eb2ab is selecting the pages with a repetitive name. However this ebook shows 1 valid page (part1.xhtml) but 5 pages named fundX.html which are not real text, causing a false positive.... I'm working to find another way to filter it.

**Comment by luky92** [comment]
sure and I dont have a problem with having the whole book including this type of metdata problem is it SKIPS the actual story

**Comment by luky92** [comment]
so for some reson part1 is totally skiped

**Comment by ROBERT-MCDOWELL** [comment]
the script cannot filter the text ifself but only the pages. as part1 is only one page, and fundX several pages so the script thinks that the real text is from fundX since a book usually have more than one page.

**Comment by danieltomasz** [comment]
@luky92  the workaround would be to use directly other format provided by the portal (pdf of txt for example)

**Comment by luky92** [comment]
Yeah I already figured that out on the plus side this file was only for testing so no biggie and I assume normal full books will not have this problem

**Comment by danieltomasz** [comment]
the actual content has <p class="paragraph"></p> the fundrasing pages have only  <div class="fundraising">    </div>,  mnaybe some sort of heuristic for filtering is possible but this will work only for this  particular portal

**Comment by ROBERT-MCDOWELL** [comment]
@danieltomasz ok but understand that the millions ebook have their own class, own pages, own xhtml mark etc...
there is absolutely no standards followed by epub standard consortium (unless epub3 that started a few), 
it's a complete mess of 16 years of ebook....

**Comment by luky92** [comment]
Now that I read up on it I'm telling you from experience with parsing pdfs before don't try to do any kind of fancy filtering or heuristics here the best you can do if you want is display pages to the user as a list with preview and let them choose which ones should be read otherwise you are going to have bugs about this sort of thing all the time

**Comment by ROBERT-MCDOWELL** [comment]
you forgot here it's TTS (text to speech)!, tell me how to let the user choose which text to listen, creating hundreds of index on the 
audio file? with the text on each index? how a blind person will choose from it?

**Comment by luky92** [comment]
That's the easy part at least for pdfs and from what I understand epubs as
well you just display to the user the list of pages you can read to choose
you can let them choose the order too and by default just read everything
always

niedz., 29 gru 2024, 00:28 użytkownik ROBERT MCDOWELL <
***@***.***> napisał:

**Comment by ROBERT-MCDOWELL** [comment]
you are talking about something that has nothing to do with the issue we have here...
an audiobook is AUDIO, some listeners will use it with an AUDIO player, no screen no keyboard. that's why the challenge is much complex than filter/parse a pdf.....

**Comment by luky92** [comment]
Oh different goals yep yep I assumed that we are talking about something
that will just be an ebook2audiobook converter so that I can listen to them
later in another player but if we are talking about staying in the same app
I don't see a way to skip the legalese or some other parts of the book you
could do it automatically with the length of the page but that's not
foolproof at all

niedz., 29 gru 2024, 00:43 użytkownik ROBERT MCDOWELL <
***@***.***> napisał:

**Comment by ROBERT-MCDOWELL** [comment]
there is always a way, but not 100% bullet proof. the most complex would be to add another A.I. with a model that recognizes a boo text than legal and technical text but in +1124 languages.... forget it....

**Comment by luky92** [comment]
Thinking about it maybe you could use a text LLM to help with that something small like one of the phi Microsoft models or something like that just give it a snippet of each page in the file and ask it if it's an attribution text or something that is a part of the main book now granted you would have to make sure you are giving it enough to assess it but it could be something

**Comment by ROBERT-MCDOWELL** [comment]
you will reach maybe 3 to 5% accuracy than text length. text length is still not a solution, I saw poems book with 2 sentences per page... and a legal printer stuff with more than 1000 chars.... you can search by keyword indeed, but with so many languages it's another project into the project.

**Comment by ROBERT-MCDOWELL** [comment]
ok I made test on your ebook with my last update and it seems to work now..
you will confirm when you will test on your side and close this issue then.

**Comment by ROBERT-MCDOWELL** [comment]
should be fixed in the last git.
